xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h

```diff
@@ -645,7 +1015,129 @@ inline void SetSettingFloat(const std::string& settingName, float settingValue)
+///
+///
+/// ----------------------------------------------------------------------------
+///
+/// **Example:**
+/// ~~~~~~~~~~~~~{.cpp}
+/// #include <kodi/General.h>
+///
+/// enum myEnumValue
+/// {
+///   valueA,
+///   valueB,
+///   valueC
+/// };
+///
+/// myEnumValue value;
+/// if (!kodi::CheckSettingEnum<myEnumValue>("my_enum_value", value))
+///   value = valueA; // My default of them
+/// ~~~~~~~~~~~~~
+///
+template<typename enumType>
+inline bool CheckSettingEnum(const std::string& settingName, enumType& settingValue)
+{
+  using namespace kodi::addon;
+
+  int settingValueInt = static_cast<int>(settingValue);
+  bool ret = CAddonBase::m_interface->toKodi->get_setting_int(
+      CAddonBase::m_interface->toKodi->kodiBase, settingName.c_str(), &settingValueInt);
+  if (ret)
+    settingValue = static_cast<enumType>(settingValueInt);
+  return ret;
+}
+//------------------------------------------------------------------------------
+
+//==============================================================================
+/// @brief Get enum setting value.
+///
+/// The setting name relate to names used in his <b>settings.xml</b> file.
+///
+/// @param[in] settingName The name of asked setting
+/// @return The value of setting, forced to <b>`0`</b> if not found;
+///
+/// @remark The enums are used as integer inside settings.xml.
+///
+///
+/// ----------------------------------------------------------------------------
+///
+/// **Example:**
+/// ~~~~~~~~~~~~~{.cpp}
+/// #include <kodi/General.h>
+///
+/// enum myEnumValue
+/// {
+///   valueA,
+///   valueB,
+///   valueC
+/// };
+///
+/// myEnumValue value = kodi::GetSettingEnum<myEnumValue>("my_enum_value");
+/// ~~~~~~~~~~~~~
+///
+template<typename enumType>
+inline enumType GetSettingEnum(const std::string& settingName)
+{
+  enumType settingValue = static_cast<enumType>(0);
+  CheckSettingEnum(settingName, settingValue);
+  return settingValue;
+}
+//------------------------------------------------------------------------------
+
+//==============================================================================
+/// @brief Set enum setting of addon.
+///
+/// The setting name relate to names used in his <b>settings.xml</b> file.
+///
+/// @param[in] settingName The name of setting
+/// @param[in] settingValue The setting value to write
+///
+/// @remark The enums are used as integer inside settings.xml.
+///
+///
+/// ----------------------------------------------------------------------------
+///
+/// **Example:**
+/// ~~~~~~~~~~~~~{.cpp}
+/// #include <kodi/General.h>
+///
+/// enum myEnumValue
+/// {
+///   valueA,
+///   valueB,
+///   valueC
+/// };
+///
+/// myEnumValue value = valueA;
+/// kodi::SetSettingEnum<myEnumValue>("my_enum_value", value);
+/// ~~~~~~~~~~~~~
+///
+template<typename enumType>
+inline void SetSettingEnum(const std::string& settingName, enumType settingValue)
+{
+  using namespace kodi::addon;
```

Comment:
Nope. Never use 'using namespace' in a header file, as this can lead to all kind of weird errors when this header gets included together with other headers.